src/main/java/com/aws/greengrass/dependency/Context.java

```diff
@@ -563,7 +580,8 @@ public final <E extends Exception> T computeObjectIfEmpty(
                 if (object != null) {
                     return object;
                 }
-
+                // Mapping function should make sure that the object is not injected already. Otherwise, it results in
```

Comment:
If object is null then that means it is not injected already right?

Reply:
Yes. This comment is for the mapping function in line 585 as we're injecting the object in the same line.

Reply:
We're holding the lock so the object will not change, and we know the object is null at this point (and thus not injected already), so the mappingFunction can safely assume the object is not injected already. This comment says the opposite

Reply:
The `object` here is not same the as the object created and injected already by the mapping function. Mapping function creates the object and injects it. Now we made changes to not.

Reply:
I see that now. Sorry for the back and forth on a comment

Not a big deal, approving now; but for the javadoc change discussed in standup, I meant it should be noted the javadoc spec of the mappingFunction provided to computeObjectIfEmpty, as that is part of the API that was being misused (when it was being passed a function that called newInstance that did also inject)